vcd/datasource_vcd_nsxt_segment_qos_profile.go

```diff
@@ -0,0 +1,143 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdNsxtSegmentQosProfile() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceNsxtSegmentQosProfileRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of Segment QoS Profile",
+			},
+			"nsxt_manager_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"nsxt_manager_id", "vdc_id", "vdc_group_id"},
+				Description:  "ID of NSX-T Manager",
+			},
+			"vdc_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"nsxt_manager_id", "vdc_id", "vdc_group_id"},
+				Description:  "ID of VDC",
+			},
+			"vdc_group_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"nsxt_manager_id", "vdc_id", "vdc_group_id"},
+				Description:  "ID of VDC Group",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of Segment QoS Profile",
+			},
+			"class_of_service": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Groups similar types of traffic in the network and each type of traffic is treated as a class with its own level of service priority",
+			},
+			"dscp_priority": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Differentiated Services Code Point priority",
+			},
+			"dscp_trust_mode": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Differentiated Services Code Point trust mode",
+			},
+			"egress_rate_limiter_avg_bandwidth": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Average bandwidth in Mb/s",
```

Comment:
Just to confirm, these bandwidth fields are all in mega*bits* (Mb)?

Reply:
I have trusted docs for it. Will recheck.

Reply:
This looks to be correct when double-checking in VCD docs as well as NSX-T manager itself.
NSX-T manager has `Mbps`, which is the same as `MB/s` (Megabits per second)

Reply:
Great! Thank you!